source/extensions/quic_listeners/quiche/platform/quic_mem_slice_storage_impl.h

```diff
@@ -20,26 +20,29 @@ class QuicMemSliceStorageImpl {
   QuicMemSliceStorageImpl(const iovec* iov, int iov_count, QuicBufferAllocator* allocator,
                           const QuicByteCount max_slice_len);
 
-  QuicMemSliceStorageImpl(const QuicMemSliceStorageImpl& other) { buffer_.add(other.buffer_); }
+  QuicMemSliceStorageImpl(const QuicMemSliceStorageImpl& other) { *this = other; }
 
   QuicMemSliceStorageImpl& operator=(const QuicMemSliceStorageImpl& other) {
     if (this != &other) {
-      if (buffer_.length() > 0) {
-        buffer_.drain(buffer_.length());
+      for (auto& mem_slice : other.mem_slices_) {
+        Envoy::Buffer::OwnedImpl buffer;
+        buffer.add(mem_slice.data(), mem_slice.length());
+        mem_slices_.emplace_back(buffer, mem_slice.length());
```

Comment:
I think making copies is fine since it is only needed in tests, but worth adding a comment.

Reply:
Fair enough.